### PR TITLE
feat(mint): chain reset-resistant mint schedule

### PR DESCRIPTION
**Pull request by gsora**
This PR addresses the issue where if for some reason we have to restart the chain from block 0, the minting schedule would be restarted as well.

Instead of calculating the current year based on the block height, write down the first ever execution of the network and do calculations that way.

### CODE REVIEW

File: x/mint/keeper/keeper.go [hunk 1]
```diff
@@ -62,6 +65,80 @@ func (k Keeper) SetParams(ctx sdk.Context, params types.Params) {
 	k.paramSpace.SetParamSet(ctx, &params)
 }
 
+// SetBootstrapDateCanary sets the bootstrap date canary boolean to true.
+// If overwrite is true, regardless of whether or not the canary was already set, it will be set again.
```
Comment (barriebyron) [inline x/mint/keeper/keeper.go]:
```suggestion
// If overwrite is true, the date is set again even when it was already set.
```

**Review by barriebyron — COMMENTED** [review]
@gsora I always like to read the code comments to see if the logic is clear to me, a non-developer... perhaps this wording suggestion makes the comment more understandable

cheerio!

**Comment by barriebyron** [comment]
hi team! what's the status of this PR? (I hope that my suggested changes did not interrupt?)

**Review by noandrea — APPROVED** [review]
LGTM :100:

**Comment by noandrea** [comment]
for reference, I've run the tests, simulations and smoke test querying supply.

**Pull request merged**